File: app/src/main/java/com/akay/fitnass/data/RunsRepository.java
package com.akay.fitnass.data;

import android.arch.lifecycle.LiveData;

import com.akay.fitnass.data.model.ActiveRuns;
import com.akay.fitnass.data.model.Runs;

import java.util.List;

public interface RunsRepository {

    ActiveRuns getActiveRuns();

    LiveData<ActiveRuns> getLiveActiveRuns();

    LiveData<List<Runs>> getLiveRuns();

    Runs getById(long id);

    void upsertActiveRuns(ActiveRuns activeRuns);

    void deleteActiveRuns();

    void saveRuns(Runs runs);
}

File: app/src/main/java/com/akay/fitnass/data/RunsRepository.kt
package com.akay.fitnass.data

import androidx.lifecycle.LiveData

import com.akay.fitnass.data.model.ActiveRuns
import com.akay.fitnass.data.model.Runs

interface RunsRepository {

    fun getActiveRuns(): ActiveRuns

    fun getLiveActiveRuns(): LiveData<ActiveRuns>

    fun getLiveRuns(): LiveData<List<Runs>>

    fun getById(id: Long): Runs

    fun upsertActiveRuns(activeRuns: ActiveRuns)

    fun deleteActiveRuns()

    fun saveRuns(runs: Runs)

    fun deleteRuns(runs: Runs)
}
